Moonscraper Chart Editor/Assets/Scripts/Game/Charts/IO/Chart/ChartReader.cs

```diff
@@ -859,7 +891,27 @@ static void GroupAddFlags(IList<Note> notes, Note.Flags flag, int index, int len
         {
             for (int i = index; i < index + length; ++i)
             {
-                notes[i].flags = notes[i].flags | flag;
+                AddNoteFlag(notes[i], flag);
+            }
+        }
+
+        static void AddNoteFlag(Note note, Note.Flags flag)
+        {
+            // Don't add if the flag to be added is lower-priority than a conflicting, already-added flag
+            Note.Flags prioritizedFlag;
+            if (ChartIOHelper.c_noteFlagIgnoreLookup.TryGetValue(flag, out prioritizedFlag))
+            {
+                if ((note.flags & prioritizedFlag) != Note.Flags.None)
+                    return;
+            }
+
+            note.flags |= flag;
+
+            // Remove any flags that the newly-added flag should override
+            Note.Flags flagToRemove;
+            if (ChartIOHelper.c_noteFlagOverrideLookup.TryGetValue(flag, out flagToRemove))
+            {
+                note.flags &= ~flagToRemove;
```

Comment:
The point of doing dictionary lookups here was to keep things simpler and not have to code everything in everywhere else. There are multiple places where I'd have to add additional parameters if I make the suggested change; doing it all in one place keeps things self-contained.

Reply:
Yes, but the tradeoff is that it makes that code much more readable. As it currently stands, the function has a hidden side-effect which can make future debugging a pain. As far as I can tell, the addition places it would need to be added to would just be in the process fn dictionaries, which is fine imo cause it's an obvious config definition.
It's also inefficient as it has to do 1-2 dictionary lookups for every flag that it has to process when it could be more efficient by pre-computing this.

Reply:
You can do something like 
```
Note.Flags ghostNoteFlags = Note.Flags.ProDrums_Ghost;
Note.Flags ghostBlockingFlags; ChartIOHelper.c_noteFlagIgnoreLookup.TryGetValue(ghostNoteFlags , out ghostBlockingFlags);
Note.Flags ghostOverrideFlags; ChartIOHelper.c_noteFlagOverrideLookup.TryGetValue(ghostNoteFlags , out ghostBlockingFlags);
...
{ c_drumsGhostOffset + 1, ghostNoteFlags, ghostBlockingFlags, ghostOverrideFlags  },
```
in the BuildDictionaries function, and then it's still reliant on a single point in the config.

Reply:
Could even go further and pack those 3 flag parameters into a class and then pass that class around
`{ c_drumsGhostOffset + 1, ghostFlagParams  },`
Whereby the class builds the blocking and override flags based on the dictionary.

Reply:
> in the BuildDictionaries function

The .chart parser doesn't have any dictionary builder functions, I'll have to add one for drums to do that.

Reply:
So close, yet so far...
![image](https://user-images.githubusercontent.com/29052821/187604017-38bdb7e3-fc6b-42cb-a0a1-6904868fb585.png)

Reply:
Got things figured out, works good now. Hopefully it's all good to you.

Reply:
> > in the BuildDictionaries function
> 
> The .chart parser doesn't have any dictionary builder functions, I'll have to add one for drums to do that.

I was thinking of the MidReader in my head when I wrote that. I actually don't like the build dictionary functions in general since it makes maintenance on them more of a pain compared to just explicitly defining a function for each explicit number, and I plan to remove that in the eventual refactor stuff I'm doing. 
Shouldn't need to add a build dictionaries function anyway. Could just remove the dictionary and just replace it with a lookup function instead. Then you wouldn't run into the issue of trying to access the dictionary during a static initialisation of something else.

Reply:
In hindsight I never really needed the builder function in the first place, I can just use the static NoteFlagPriority properties for accents/ghosts/cymbals in their respective keys in place of what was originally just the note flags.